### PR TITLE
Added the logic in queue.js to show the queue name and location

### PR DESCRIPTION
**Pull request by sgorse**

### CODE REVIEW

File: pages/queue.js [hunk 1]
```diff
@@ -90,6 +95,7 @@ Queue.propTypes = {
 const mapStateToProps = (state, ownProps) => ({
   isFetching: state.queues.isFetching,
   hasQueue: !!state.queues.queues[ownProps.queueId],
+  queueInfo: state.queues.queues[ownProps.queueId],
```
Comment (nwalters512) [inline pages/queue.js]: Should be named `queue` for consistency with the rest of the application.

File: pages/queue.js [hunk 2]
```diff
@@ -46,7 +47,8 @@ class Queue extends React.Component {
   }
 
   render() {
-    const { isFetching, hasQueue } = this.props
+    const { isFetching, hasQueue, queueInfo } = this.props
+    const location = this.props.queueInfo.location || 'Not Available'
```
Comment (nwalters512) [inline pages/queue.js]: `queueInfo` might not be defined while the page is still loading (try refreshing the queue page with your code as-is). Avoid accessing any properties until after the various `isFetching` checks.

File: pages/queue.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ import QuestionNotificationsToggle from '../components/QuestionNotificationsTogg
 
 class Queue extends React.Component {
   static getInitialProps({ isServer, store, query }) {
+    console.log(this)
```
Comment (nwalters512) [inline pages/queue.js]: Remove extraneous log.

File: pages/queue.js [hunk 4]
```diff
@@ -56,6 +58,9 @@ class Queue extends React.Component {
     return (
       <Layout>
         <Container fluid>
+          <h5>
```
Comment (nwalters512) [inline pages/queue.js]: Try splitting into two lines, maybe with a combination of `<h3>`/`<h5 className="text-muted">`, would look even better with the location icon that's used on the queue cards on the homepage!

**Comment by nwalters512** [comment]
You should use `PropTypes.shape` instead, take a look at the `course` prop on the course page.

**Comment by sgorse** [comment]
Ok, thanks for pointing that out. I am still getting a warning on the client console for the queue.id. I am getting this warning along with other warnings that do not pertain to the queue. The one that does is "Warning: Failed prop type: Invalid prop `queue.id` of type `number` supplied to `Queue`, expected `object`". Am I not setting the queue.id propType properly here?
```
Queue.propTypes = {
  isFetching: PropTypes.bool.isRequired,
  hasQueue: PropTypes.bool.isRequired,
  fetchQueue: PropTypes.func.isRequired,
  queueId: PropTypes.number.isRequired,
  dispatch: PropTypes.func.isRequired,
  queue: PropTypes.objectOf(
    PropTypes.shape({
      id: PropTypes.number,
      name: PropTypes.string,
      location: PropTypes.location,
    })
  ),
}
```

**Comment by nwalters512** [comment]
You don't need the `objectOf` part, that's only if you have an object whose values are all queues. Just `shape` is enough. The other bit is not related to you: https://github.com/illinois/queue/issues/79

File: pages/queue.js [hunk 5]
```diff
@@ -47,15 +49,22 @@ class Queue extends React.Component {
...
     }
     if (!isFetching && !hasQueue) {
       return <Error statusCode={404} />
     }
+    const locationText = this.props.queue.location || 'No location specified'
     return (
       <Layout>
         <Container fluid>
+          <h3>{this.props.queue.name}</h3>
+          <h5 className="text-muted">
```
Comment (nwalters512) [inline pages/queue.js]: Can you add a bottom margin here? Maybe `mb-4`.

Comment (sgorse) [reply pages/queue.js]:
```
         <Col
            xs={{ size: 12 }}
            md={{ size: 4 }}
            lg={{ size: 3 }}
            className="mb-3"
          >
            <h3>{this.props.queue.name}</h3>
            <h5 className="text-muted">
              <FontAwesomeIcon icon={faMapMarker} fixedWidth className="mr-2" />
              {locationText}
            </h5>
          </Col>
```
Would this be how I add the bottom margin?

Comment (nwalters512) [reply pages/queue.js]: You can just add it directly to the `h5`

**Pull request merged**